src/wallet/wallet.cpp

```diff
@@ -842,6 +842,25 @@ bool CWallet::IsSpentKey(const uint256& hash, unsigned int n) const
     return false;
 }
 
+void CWallet::SetUsedAddressState(WalletBatch& batch, const CTxOut& cout, bool used)
+{
+    AssertLockHeld(cs_wallet);
+    CTxDestination dst;
+    if (ExtractDestination(cout.scriptPubKey, dst) && IsMine(dst)) {
+        if (used && !GetDestData(dst, "address_used", nullptr)) {
+            AddDestData(batch, dst, "address_used", "p"); // p for "present", opposite of absent (null)
+        } else if (!used && GetDestData(dst, "address_used", nullptr)) {
+            EraseDestData(batch, dst, "address_used");
+        }
+    }
+}
+
+bool CWallet::IsUsedAddress(const CTxDestination& dst) const
+{
+    AssertLockHeld(cs_wallet);
+    return GetDestData(dst, "address_used", nullptr);
```

Comment:
An existing wallet will not have these set, ignore this fact? Or just add a release note saying it that a rescan fixes this?

Reply:
Yes I think the best solution would be adding a release note as you suggested, I will look into doing this shortly